### PR TITLE
Add stroke type/line style to default tool settings

### PR DESCRIPTION
**Pull request by philwilei**
I've been sitting on this change for while. I saw that someone else requested this feature in #4815, so I decided to finally complete it.
Here's the preferences dialog with the new combobox. It will only show when "Pen" is selected as the default tool, since this is the only tool that supports different line styles AFAIK:
image

Suggestions/improvements welcome as this is my first PR on this project and I have limited C++ experience :smile:

### CODE REVIEW

**Comment by bhennion** [comment]
Thanks for the PR! The changes look reasonable. There is a clang-format issue though. 
Would you mind running `git clang-format` on your branch?

**Comment by philwilei** [comment]
Thank you for pointing it out. I ran `git clang-format master` and squashed the fixed formatting into my original commits.

**Review by rolandlo — COMMENTED** [review]
Thanks for the valuable feature. The code looks good to me and it works fine. Ideally we would want to have the stroke type also be configurable for stylus buttons (Stylus tab in the preferences). Would you like to add that in this PR as well? Otherwise we can also postpone that to another PR.

**Comment by philwilei** [comment]
I just looked into implementing this for the stylus buttons and it wasn't obvious to me where I would need to start, but I am interested in getting myself more familiar with the codebase and finishing this feature. I think it's best to put these additional changes into this PR, so merging will have to wait while I figure this out :smiley:

**Comment by philwilei** [comment]
The feature now also works for stylus buttons. All I had to do was to add an if statement and create `ToolHandler::setButtonStrokeType`. clang-format changed lots of lines in `ToolHandler.cpp`, so the commit looks a lot more involved than it is.

**Review by rolandlo — CHANGES_REQUESTED** [review]
Thanks for implementing stroke type for stylus buttons.
I found two issues with assigning a stylus button to the pen:
- `Stroke Type - don't change` doesn't work properly.
- for `Drawing Type` anything below `Draw Arrow` doesn't work, apparently since the `Draw Double Arrow` is missing. That's not your mistake, but would you fix it anyway?

We have agreed on avoiding formatting changes unrelated to the PR. So please undo those formatting changes in `ToolHandler.cpp`. Running `git clang-format --binary=clang-format-$VERSION --diff $BASE_COMMIT` should only format the changes you made, not anything unrelated.

File: ui/settingsButtonConfig.glade [hunk 1]
```diff
@@ -133,19 +135,65 @@
...
             </child>
             <child>
               <placeholder/>
             </child>
+            <child>
+              <placeholder/>
+            </child>
+            <child>
+              <placeholder/>
+            </child>
```
[the remark this answers is not in the document]
Comment (philwilei) [reply ui/settingsButtonConfig.glade]: Adressed in most recent push.

**Comment by philwilei** [comment]
I have finally gotten around to working on this feature again. I rebased onto current `master`, addressed the requested changes and fixed the issues my implementiation still had. I just noticed that the issue with "Draw Double Arrow" has already been fixed separately for the 1.2 release; if preferred I can remove this from my branch and force-push again.

**Review by rolandlo — APPROVED** [review]
LGTM. Everything seems to work fine now. I suggest to leave the (duplicated) double arrow related changes away though, since it will mess with the commit history otherwise.

**Comment by rolandlo** [comment]
Merging in 48 hours if no objections are raised.

**Comment by rolandlo** [comment]
Thanks @philwilei for your contribution. It's a nice improvement. I just merged it.

**Pull request merged**